File: Data Structures & Algorithms Mastery/Graphs/Simple Graph/Java/Simple Graph/src/Edge.java
public class Edge {
    Vertex src;
    Vertex dest;
    int weight;

    public Edge() {}

    public Edge(Vertex src, Vertex dest) {
        this.src = src;
        this.dest = dest;
        this.weight = 1;
    }

    public Edge(Vertex src, Vertex dest, int weight) {
        this.src = src;
        this.dest = dest;
        this.weight = weight;
    }

    public void setSrc(Vertex src) {
        this.src = src;
    }

    public Vertex getSrc() {
        return this.src;
    }

    public void setDest(Vertex dest) {
        this.dest = dest;
    }

    public Vertex getDest() {
        return this.dest;
    }

    public void setWeight(int weight) {
        this.weight = weight;
    }

    public int getWeight() {
        return this.weight;
    }
}

File: My CS Projects/NJ Transit Network/nj-transit-network/Edge.cs
namespace nj_transit_network;

public class Edge {
    private Vertex src;
    private Vertex dest;
    private double weight;

    public Edge(Vertex src, Vertex dest) {
        this.src = src;
        this.dest = dest;
        this.weight = 1.0;
    }

    public Edge(Vertex src, Vertex dest, double weight) {
        this.src = src;
        this.dest = dest;
        this.weight = weight;
    }

    public void SetSrc(Vertex src) {
        this.src = src;
    }

    public Vertex GetSrc() {
        return this.src;
    }

    public void SetDest(Vertex dest) {
        this.dest = dest;
    }

    public Vertex GetDest() {
        return this.dest;
    }

    public void SetWeight(double weight) {
        this.weight = weight;
    }

    public double GetWeight() {
        return this.weight;
    }

    override public string ToString() {
        return this.src + " -(" + this.weight + ")> " + this.dest;
    }
}